src/irmin/node.ml

```diff
@@ -363,6 +377,58 @@ struct
       | _ -> None)
 end
 
+module Make_generic_key_v2
+    (Hash : Hash.S) (Path : sig
+      type step [@@deriving irmin]
+    end)
+    (Metadata : Metadata.S)
+    (Contents_key : Key.S with type hash = Hash.t)
+    (Node_key : Key.S with type hash = Hash.t) =
+struct
+  include Make_generic_key (Hash) (Path) (Metadata) (Contents_key) (Node_key)
```

Comment:
You also need to override the portable typerepr

Reply:
thanks for catching this, should be good now